releases/release-4.0.14.md

```diff
@@ -0,0 +1,160 @@
+    - Important security issue for handling ALTER USER statements [#25347](https://github.com/pingcap/tidb/pull/25347)
+    - Fix the bug that `TIKV_REGION_PEERS` table did not have the correct `DOWN` status. [#24918](https://github.com/pingcap/tidb/pull/24918)
+    - Handle a potential statistic object's memory leak when HTTP api is used [#24650](https://github.com/pingcap/tidb/pull/24650)
+    - Don't let SPM be affected by charset [#23295](https://github.com/pingcap/tidb/pull/23295)
+    - Time: parse datatime won't truncate the reluctant string. [#22260](https://github.com/pingcap/tidb/pull/22260)
+    - Fix a bug that `select into outfile` has no data with year column type [#22185](https://github.com/pingcap/tidb/pull/22185)
+
++ TiKV
+
+    - Fix duration calculation panics on certain platforms [#10572](https://github.com/tikv/tikv/pull/10572)
+    - Fix wrong function cast double to double [#10532](https://github.com/tikv/tikv/pull/10532)
+    - Ensure panic output is flushed to the log [#10488](https://github.com/tikv/tikv/pull/10488)
+    - Avoid panic when building a snapshot twice if encryption enabled [#10462](https://github.com/tikv/tikv/pull/10462)
+    - Copr: fix the wrong arguments type of json_unquote [#10425](https://github.com/tikv/tikv/pull/10425)
+    - skip clearing callback during gracefully shutdown to avoid breaking ACID in some cases [#10395](https://github.com/tikv/tikv/pull/10395)
+    - Fix backup threads leak [#10360](https://github.com/tikv/tikv/pull/10360)
+    - Approximate split range evenly [#10275](https://github.com/tikv/tikv/pull/10275)
+    - BR now supports S3-compatible storage using virtual-host addressing style. [#10242](https://github.com/tikv/tikv/pull/10242)
+    - Limit CDC sink memory consumption. [#10147](https://github.com/tikv/tikv/pull/10147)
+    - Add metrics to see pending pd heartbeats number [#10008](https://github.com/tikv/tikv/pull/10008)
+    - Change the default merge-check-tick-interval to 2 to speed up the retry of merge process [#9676](https://github.com/tikv/tikv/pull/9676)
+    - Fix an issue that split may panic and corrupt the metadata if the split process is too slow and region merge is open. [#9584](https://github.com/tikv/tikv/pull/9584)
+    - Fix region heartbeat preventing from TiKV splitting large regions [#10274](https://github.com/tikv/tikv/pull/10274)
+    - Fix wrong statistics due to format inconsistency of CM Sketch between TiKV and TiDB [#10433](https://github.com/tikv/tikv/pull/10433)
+    - Fix wrong apply wait duration metrics [#9966](https://github.com/tikv/tikv/pull/9966)
+    - Fix "Missing Blob" error after using delete_files_in_range with titan [#10232](https://github.com/tikv/tikv/pull/10232)
+    - Shutdown status server first to report correct liveness to health check [#10504](https://github.com/tikv/tikv/pull/10504)
+    - Always respond to stale peer to destroy it quicker [#10400](https://github.com/tikv/tikv/pull/10400)
```

Comment:
## Compatibility Changes

    - Change the default merge-check-tick-interval to 2 to speed up the retry of merge process [#9676](https://github.com/tikv/tikv/pull/9676)

## Feature Enhancements

    - Add metrics to see pending pd heartbeats number [#10008](https://github.com/tikv/tikv/pull/10008)
    - BR now supports S3-compatible storage using virtual-host addressing style. [#10242](https://github.com/tikv/tikv/pull/10242)

## Improvements

    - Always respond to stale peer to destroy it quicker [#10400](https://github.com/tikv/tikv/pull/10400)
    - Shutdown status server first to report correct liveness to health check [#10504](https://github.com/tikv/tikv/pull/10504)
    - Limit CDC sink memory consumption. [#10147](https://github.com/tikv/tikv/pull/10147)
    - Approximate split range evenly [#10275](https://github.com/tikv/tikv/pull/10275)

## Bug Fixes
    - Fix duration calculation panics on certain platforms [#10572](https://github.com/tikv/tikv/pull/10572)
    - Fix wrong function cast double to double [#10532](https://github.com/tikv/tikv/pull/10532)
    - Ensure panic output is flushed to the log [#10488](https://github.com/tikv/tikv/pull/10488)
    - Avoid panic when building a snapshot twice if encryption enabled [#10462](https://github.com/tikv/tikv/pull/10462)
    - Copr: fix the wrong arguments type of json_unquote [#10425](https://github.com/tikv/tikv/pull/10425)
    - skip clearing callback during gracefully shutdown to avoid breaking ACID in some cases [#10395](https://github.com/tikv/tikv/pull/10395)
    - Fix backup threads leak [#10360](https://github.com/tikv/tikv/pull/10360)
    - Fix an issue that split may panic and corrupt the metadata if the split process is too slow and region merge is open. [#9584](https://github.com/tikv/tikv/pull/9584)
    - Fix region heartbeat preventing from TiKV splitting large regions [#10274](https://github.com/tikv/tikv/pull/10274)
    - Fix wrong statistics due to format inconsistency of CM Sketch between TiKV and TiDB [#10433](https://github.com/tikv/tikv/pull/10433)
    - Fix wrong apply wait duration metrics [#9966](https://github.com/tikv/tikv/pull/9966)
    - Fix "Missing Blob" error after using delete_files_in_range with titan [#10232](https://github.com/tikv/tikv/pull/10232)